contrib/99designs/gqlgen/tracer.go

```diff
@@ -0,0 +1,96 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+// Package gqlgen contains an implementation of a gqlgen tracer, and functions to construct and configure the tracer.
+// The tracer can be passed to the gqlgen handler (see package github.com/99designs/gqlgen/handler)
+package gqlgen
+
+import (
+	"context"
+	"fmt"
+	"math"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"github.com/99designs/gqlgen/graphql"
+)
+
+type trace struct {
+	graphql.NopTracer
+	serviceName   string
+	analyticsRate float64
+}
+
+// New creates an a graphql.Tracer instance that can be passed to a gqlgen handler.
+// Options can be passed in for further configuration.
+func New(opts ...Option) graphql.Tracer {
+	t := &trace{}
+	defaults(t)
+	for _, o := range opts {
+		o(t)
+	}
+	return t
+}
+
+func (t *trace) StartOperationExecution(ctx context.Context) context.Context {
+	rctx := graphql.GetRequestContext(ctx)
+	opName := ""
```

Comment:
Can we use `name`?

If we want to [declare a variable](https://dave.cheney.net/2014/05/24/on-declaring-variables) at its [zero value](https://golang.org/ref/spec#The_zero_value) prefer:
```go
var name string // ""
var b bool // false
var d int // 0
```
and not
```go
name := ""
b := false
d := 0
```

Reply:
Yes, I remember reading this.